packages/core/src/storage/MessageRepository.ts

```diff
@@ -1,6 +1,6 @@
 import type { EncryptedMessage } from '../types'
 
 export interface MessageRepository {
-  takeFromQueue(connectionId: string, limit?: number): EncryptedMessage[]
-  add(connectionId: string, payload: EncryptedMessage): void
+  takeFromQueue(connectionId: string, limit?: number): EncryptedMessage[] | Promise<EncryptedMessage[]>
+  add(connectionId: string, payload: EncryptedMessage): void | Promise<void>
```

Comment:
Can this still be sync? I thought all repository operations are async.

Reply:
The message repository is a special type of repository. We should probably rework this in the near future if we want to start using AFJ as a mediator. For now this makes it possible to make a custom repo implementation based on the indy wallet storage (which is indeed async).

Reply:
It should probably be called something like queue (as that's what is is based on the methods e.g. takeFromQueue)